protocol/message_persistence.go

```diff
@@ -1131,3 +1150,78 @@ func (db sqlitePersistence) InvitationByID(id string) (*GroupChatInvitation, err
 		return nil, err
 	}
 }
+
+// ClearHistory deletes all the messages for a chat and updates it's values
+func (db sqlitePersistence) ClearHistory(chat *Chat, currentClockValue uint64) (err error) {
+	var tx *sql.Tx
+
+	tx, err = db.db.BeginTx(context.Background(), &sql.TxOptions{})
+	if err != nil {
+		return
+	}
+
+	defer func() {
+		if err == nil {
+			err = tx.Commit()
+			return
+		}
+		// don't shadow original error
+		_ = tx.Rollback()
+	}()
+	err = db.clearHistory(chat, currentClockValue, tx)
+
+	return
+}
+
+// Deactivate chat sets a chat as inactive and clear its history
+func (db sqlitePersistence) DeactivateChat(chat *Chat, currentClockValue uint64) (err error) {
+	var tx *sql.Tx
+
+	tx, err = db.db.BeginTx(context.Background(), &sql.TxOptions{})
+	if err != nil {
+		return
+	}
+
+	defer func() {
+		if err == nil {
+			err = tx.Commit()
+			return
+		}
+		// don't shadow original error
+		_ = tx.Rollback()
+	}()
+	err = db.deactivateChat(chat, currentClockValue, tx)
+
+	return
+}
+
+func (db sqlitePersistence) deactivateChat(chat *Chat, currentClockValue uint64, tx *sql.Tx) error {
+	chat.Active = false
+	err := db.saveChat(tx, *chat)
+	if err != nil {
+		return err
+	}
+
+	return db.clearHistory(chat, currentClockValue, tx)
+}
+
+func (db sqlitePersistence) clearHistory(chat *Chat, currentClockValue uint64, tx *sql.Tx) error {
+	// Set deleted at clock value
```

Comment:
That's not a very helpful comment, we can see in this line that's what happens:
```go
chat.DeletedAtClockValue = currentClockValue
```
But why does it need to be set? That's what the comment should say.